### ISSUE TITLE
DAX Studio does not return identical result as a same query via pbix. Potentially due to high values

### ISSUE THREAD

**Issue by jasiluka** [DaxStudio/DaxStudio#999]
DAX Studio v3.0.5.767
Connection: AzureAnalysisServices - 16.0.51.25

## Summary of Issue
on a pbix i built a small table from calculated table. it has 4 rows with 3 columns. 
3rd column has a high value of hundreds of millions.
When using performance analyser to copy the query into dax studio, the table is identical however last column shows all blanks instead of those same values as in pbix

![image](https://user-images.githubusercontent.com/53556730/204254184-de7c9242-3c68-4508-bdaa-4dedd2606687.png)
![image](https://user-images.githubusercontent.com/53556730/204254238-cb05a197-879d-4bc9-a3e5-f6de8a7d32f1.png)

I suspect this is because DAX Studio cant display such high values?

**Comment by jasiluka** [comment]
fyi this is a query given via performance analyser:



// DAX Query
DEFINE
  VAR __DS0FilterTable = 
    TREATAS(
      {"K06M_K06M_1997_IP01_BOLI_G_AND_M_BANK_PL01"},
      'Prophet_PVFCF'[Sub_Unit_of_Account]
    )

  VAR __DS0FilterTable2 = 
    TREATAS({"PREM"}, 'Prophet_PVFCF'[Amount_Type])

  VAR __DS0Core = 
    CALCULATETABLE(
      SUMMARIZE(
        'Prophet_PVFCF',
        'Prophet_PVFCF'[Sub_Unit_of_Account],
        'Prophet_PVFCF'[Amount_Type],
        'Prophet_PVFCF'[(PVFCF_Prophet) Assumptions Update (Expenses)]
      ),
      KEEPFILTERS(__DS0FilterTable),
      KEEPFILTERS(__DS0FilterTable2)
    )

  VAR __DS0PrimaryWindowed = 
    TOPN(
      501,
      __DS0Core,
      'Prophet_PVFCF'[Sub_Unit_of_Account],
      1,
      'Prophet_PVFCF'[Amount_Type],
      1,
      'Prophet_PVFCF'[(PVFCF_Prophet) Assumptions Update (Expenses)],
      1
    )

EVALUATE
  __DS0PrimaryWindowed

ORDER BY
  'Prophet_PVFCF'[Sub_Unit_of_Account],
  'Prophet_PVFCF'[Amount_Type],
  'Prophet_PVFCF'[(PVFCF_Prophet) Assumptions Update (Expenses)]

**Comment by dgosbell** [comment]
Thanks for logging this issue. The size of the values in this column is not the issue. The problem is the column name starts with a ( character, which is confusing the databinding for the grid control. That character is OK anywhere else in the column name as long as it's not the first character which is maybe why it has not been reported before.

I have a fix for this which will go out in the next release which handles this plus the existing special characters which I was already dealing with

![image](https://user-images.githubusercontent.com/8241830/204429234-ffa2fa33-5c23-41fc-8e8c-d8495a13f2d3.png)

**Comment by pbi-ljasiuke** [comment]
thanks for the reply. however if i add a column with identical naming convention, which has lower values, they show up. Any explanation why would that be? In theory as per your explanation i would expect Prophet_PVFCF[(PVFCF_Prophet) Interest Accretion BoP] to show all blanks as well.

![image](https://user-images.githubusercontent.com/53556730/204489279-a84c602e-2a74-4bf7-8c98-6950f6a07b38.png)

**Comment by dgosbell** [comment]
> thanks for the reply. however if i add a column with identical naming convention, which has lower values, they show up. Any explanation why would that be?

It's not an identical convention, the first one has parenthesis as the first and last character and it looks like the binding only breaks if both the first and last character are parenthesis. I can **guarantee** that it's nothing to do with the data values. If you temporarily rename your first column and just delete either the opening or closing parenthesis the query should return results fine.

If you want to get technical, I followed the escaping documentation for binding path properties in WPF here:
https://learn.microsoft.com/en-us/dotnet/desktop/wpf/advanced/propertypath-xaml-syntax?redirectedfrom=MSDN&view=netframeworkdesktop-4.8#escapes-for-property-path-strings

But that does not mention parenthesis being an issue in indexing properties. My only guess is that the binding engine in WPF is trying to interpret the parenthesis as an attached property syntax. So it's probably either a bug in the Microsoft WPF framework or an omission in the documentation.

**Comment by pbi-ljasiuke** [comment]
ok thanks. clear. from your first reply i understood that only the first
character is a problem.
so it seems the problem is objects that are starting and ending with a
special character.

Thanks, it helps.

**Comment by dgosbell** [comment]
> so it seems the problem is objects that are starting and ending with a special character.

Not quite. Most special characters are OK and I already have code in place to deal with the documented special characters. It appears to be very specific to where the column name starts with `(` and also ends with `)` and I've now added a fix for this which will be part of the next release.
